st2common/st2common/models/api/execution.py

```diff
@@ -152,6 +152,10 @@ class ActionExecutionAPI(BaseAPI):
     @classmethod
     def from_model(cls, model, mask_secrets=False):
         doc = cls._from_model(model, mask_secrets=mask_secrets)
+
+        import json
+        doc['result'] = json.loads(doc['result'])
```

Comment:
I have to add this to get the result in JSONDictEscapedFieldCompatibilityField with ujson/cjson backend to work properly.  This is because _from_model will call to_mongo which dumps the result in JSON to string.  The CLI output will be displayed as a string.